doc/source/whatsnew/v0.25.0.rst

```diff
@@ -122,7 +122,7 @@ Bug Fixes
 ~~~~~~~~~
 - Bug in :func:`to_datetime` which would raise an (incorrect) ``ValueError`` when called with a date far into the future and the ``format`` argument specified instead of raising ``OutOfBoundsDatetime`` (:issue:`23830`)
 - Bug in an error message in :meth:`DataFrame.plot`. Improved the error message if non-numerics are passed to :meth:`DataFrame.plot` (:issue:`25481`)
--
+- Bug in ``read_csv`` which would not raise ``ValueError`` if a column index in ``usecols`` was out of bounds (:issue:`25623`)
```

Comment:
Should we raise an IndexError instead? Though I suppose a ValueError is consistent with the behavior of label `usecols`.

Reply:
@TomAugspurger `_validate_usecols_names` would raise a `ValueError` if `usecols` contained integer indices and was not equal to the number of columns (`names`) in length. For consistency, it now raises `ValueError` if an index is out bounds when `usecols` has length equal to the number of columns (`names`).

Reply:
I agree with using `ValueError` here.